src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.tsx

```diff
@@ -17,27 +17,42 @@ import RepositoryRow from 'app/components/repositoryRow';
 import {Panel, PanelBody, PanelHeader} from 'app/components/panels';
 import space from 'app/styles/space';
 import {t} from 'app/locale';
+import {Integration, Repository} from 'app/types';
 
-export default class IntegrationRepos extends AsyncComponent {
+type Props = AsyncComponent['props'] & {
+  integration: Integration;
+};
+
+type State = AsyncComponent['state'] & {
+  adding: boolean;
+  dropdownBusy: boolean;
+  itemList: Repository[];
+  integrationRepos: {
+    repos: {identifier: string; name: string}[];
+    searchable: boolean;
+  };
+};
+
+export default class IntegrationRepos extends AsyncComponent<Props, State> {
   static propTypes = {
     integration: PropTypes.object.isRequired,
   };
   static contextTypes = {
     organization: PropTypes.object.isRequired,
+    router: PropTypes.object,
   };
 
-  constructor(props, context) {
-    super(props, context);
-    this.state = {
-      error: false,
+  getDefaultState(): State {
+    return {
+      ...super.getDefaultState(),
       adding: false,
       itemList: [],
+      integrationRepos: {repos: [], searchable: false},
       dropdownBusy: false,
-      errors: {},
     };
   }
 
-  getEndpoints() {
+  getEndpoints(): ([string, string, any] | [string, string])[] {
     const orgId = this.context.organization.slug;
     return [
       ['itemList', `/organizations/${orgId}/repos/`, {query: {status: ''}}],
```

Comment:
@nolachen Can we add types to some of these other functions which are using an implicit `any`? VScode usually underlines things when the input is an implicit `any`. If we can do better than `any`, such as the case for `searchRepositoriesRequest` which I think we can, then we should add types.

Reply:
@scefali I got this typing straight from the base method `getEndpoints` in `asyncComponent.tsx` https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/components/asyncComponent.tsx#L347

But for some of the other functions I'll add typing to them

Reply:
@nolachen Hahah, my comment has nothing to do with `getEndpoints`. Github wouldn't let me add a comment to `searchRepositoriesRequest` as there are no code changes there and this was the closest place I could lol. What you did with `getEndpoints` is fine :)